src/workerd/jsg/jsg.c++

```diff
@@ -270,6 +271,11 @@ void Lock::terminateExecution() {
   v8Isolate->TerminateExecution();
 }
 
+void Lock::pumpMessageLoop() {
+  auto platform = IsolateBase::from(v8Isolate).getDefaultPlatform();
+  while (v8::platform::PumpMessageLoop(platform, v8Isolate)) {}
```

Comment:
This seems to be using a class method as a static method, why not use `platform.PumpMessageLoop(v8Isolate)`?
On that note, another Nit of mine is I would want to explicitly pass `MessageLoopBehavior::kDoNotWait` rather than rely on default function argument.

Another note, the `while(...)` is concerning. I guess we are still covered by the thread's watchdog..
Essentially the while loop is to solve for if someone registers another finalizer inside of the finalizers right?
Wonder if we should limit this to some const amount of loops.

Reply:
There's no `platform.PumpMessageLoop(v8Isolate)` method as part of `v8::Platform` unfortunately, it's part of `v8::DefaultPlatform` which isn't exported. 

The `while(...)` is to execute each "cleanup" task, not necessarily for the recursive use case. Although I should test how that behaves as well